service/gcs/prot/protocol.go

```diff
@@ -848,11 +848,20 @@ type MappedDirectoryV2 struct {
 	ReadOnly  bool   `json:",omitempty"`
 }
 
+// VirtualPMemMapping represents a mapped device on a given VPMem
+type VirtualPMemMapping struct {
+	DeviceOffsetInBytes int64  `json:",omitempty"`
+	DeviceSizeInBytes   int64  `json:",omitempty"`
+	MountPath           string `json:",omitempty"`
+}
+
 // MappedVPMemDeviceV2 represents a VPMem device that is mapped into a guest
 // path in the V2 schema.
 type MappedVPMemDeviceV2 struct {
 	DeviceNumber uint32 `json:",omitempty"`
 	MountPath    string `json:",omitempty"`
+	// Mapping is ignored when MountPath isn't empty
+	Mapping VirtualPMemMapping `json:",omitempty"`
```

Comment:
Feels weird to me to say that MappedVPMemDeviceV2's `MountPath` is empty when using DM but then the VirtualPMemMapping struct contains a field for `MountPath`. Could we instead remove the duplicated `MountPath` in VirtualPMemMapping and just check if `Mapping` is emtpy to know when to use DM?